WordPress/Docs/WhiteSpace/ControlStructureSpacingStandard.xml

```diff
@@ -0,0 +1,150 @@
+        <code title="Invalid: Incorrect spacing around the open and close parentheses.">
+        <![CDATA[
+// No spaces.
+while<em>(</em>have_posts()<em>)</em>{}
+
+// Too much space.
+while<em>   (   </em>have_posts()<em>   )   </em>{}
+        ]]>
+        </code>
+    </code_comparison>
+    <standard>
+    <![CDATA[
+    The open brace for the control structure must be on the same line as the close parenthesis or the control structure keyword, with one space between them.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: Open brace on the same line as the keyword/close parenthesis.">
+        <![CDATA[
+try<em> {</em>
+    // Do something.
+} catch (
+    ExceptionA | ExceptionB $e
+<em>) {</em>
+}
+        ]]>
+        </code>
+        <code title="Invalid: Open brace on a different line than the keyword/close parenthesis.">
+        <![CDATA[
+try
+<em>{</em>
+    // Do something.
+} catch ( Exception $e )
+<em>(</em>
+}
+        ]]>
+        </code>
+    </code_comparison>
+    <code_comparison>
+        <code title="Valid: One space between the keyword/close parenthesis and the open brace.">
+        <![CDATA[
+if ( $condition )<em> </em>{
+    // Do something.
+}
+        ]]>
+        </code>
+        <code title="Invalid: Too much space between the keyword/close parenthesis and the open brace.">
+        <![CDATA[
+if ( $condition )<em>     </em>{
+    // Do something.
+}
+        ]]>
+        </code>
+    </code_comparison>
+    <standard>
+    <![CDATA[
+    When using alternative control structure syntaxes, there should be one space between the close parenthesis and the colon opening the control structure body.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: One space before the colon.">
+        <![CDATA[
+foreach ( $types as $type )<em> </em>:
+    // Do something.
+endforeach;
+        ]]>
+        </code>
+        <code title="Invalid: No space before the colon.">
+        <![CDATA[
+foreach ( $types as $type )<em></em>:
+    // Do something.
+endforeach;
+        ]]>
+        </code>
+    </code_comparison>
+    <standard>
+    <![CDATA[
+    When a control structure is nested in another control structure and the closing braces follow each other, there should be no blank line between the closing braces.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: No blank line between the consecutive close braces.">
+        <![CDATA[
+if ( $a === $b ) {
+    if ( $something ) {
+        // Do something.
+    }
+}
+        ]]>
+        </code>
+        <code title="Invalid: Blank line(s) between the consecutive close braces.">
+        <![CDATA[
+if ( $a === $b ) {
+    if ( $something ) {
+        // Do something.
+    }<em>
+
+
+</em>}
+        ]]>
+        </code>
+    </code_comparison>
+    <standard>
+    <![CDATA[
+    [Optional, turned off by default]
+	There should be no blank line(s) at the start or end of the control structure body.
```

Comment:
Is the indentation on this line incorrect?

Reply:
Oh, thanks for catching that! I wonder why that wasn't caught by xmllint... Hmm.. we may need to look into that.

Reply:
Just checked - it's not checked because it would mean we'd need to add (and continuously maintain) every individual docs file in CI to allow xmllint to check this: https://github.com/WordPress/WordPress-Coding-Standards/blob/2d7f396fcc9fa726710d973b6311e26ae854f596/.github/workflows/basic-qa.yml#L79-L85

Also, we want tabs indents for rulesets, while these docs files need space indents.

We may want to look at moving the docs checks to a separate job and see about doing some bash magic or have some other type of script which loops through the docs files to xmllint them (without us having to manually maintain the list).

Reply:
I've opened an issue as a reminder: https://github.com/WordPress/WordPress-Coding-Standards/issues/2337

Reply:
I'm surprised we haven't got an `.editorconfig` file!